main.go

```diff
@@ -91,35 +99,41 @@ func runMain() error {
 		}
 
 		if !soundsDownloaded {
-			log.Println("Could not download sounds and schedule.")
+			log.Println("Could not download sounds and schedule. Will try again tomorrow")
 		}
 
 		// Wait until tomorrow.
 		playlist.WaitUntilNextDay()
-
 	}
-
 }
 
 func DownloadAndPlaySounds(audioDir string, soundCard playlist.AudioDevice) error {
-	downloader, err := NewDownloader(audioDir)
-	if err != nil {
-		return err
-	}
+	for {
+		downloader, err := NewDownloader(audioDir)
+		if err != nil {
+			return err
+		}
 
-	schedule := downloader.GetTodaysSchedule()
-	if len(schedule.Combos) == 0 {
-		return errors.New("No audio schedule for device, or no sounds to play in schedule.")
-	}
+		schedule := downloader.GetTodaysSchedule()
+		if len(schedule.Combos) == 0 {
+			return errNoSchedule
+		}
 
-	files, err := downloader.GetFilesForSchedule(schedule)
-	if err != nil {
-		return err
-	}
+		files, err := downloader.GetFilesForSchedule(schedule)
+		if err != nil {
+			return err
+		}
+
+		player := playlist.NewPlayer(soundCard, files, audioDir)
+		player.SetRecorder(AudioBaitEventRecorder{})
+		waitTime := player.TimeUntilNextCombo(schedule.Combos)
 
-	log.Printf("Playing todays audiobait schedule...")
-	player := playlist.NewPlayer(soundCard, files, audioDir)
-	player.SetRecorder(AudioBaitEventRecorder{})
-	player.PlayTodaysSchedule(schedule)
-	return nil
+		if waitTime > updateScheduleInterval {
+			time.Sleep(updateScheduleInterval)
```

Comment:
how about just returning `errNoSchedule` here so that this sleep only happens in one place. You probably want to rename `errNoSchedule` to `errTryLater` or something (and change the error text)